### PR TITLE
Correct typo in SSD documentation

### LINKED ISSUE #12807: Typo in SSD documentation
### Proposed documentation enhancement

Instead of "Spectro-Spatial Decomposition" it should be "spatio-spectral decomposition" 

https://mne.tools/dev/auto_examples/decoding/ssd_spatial_filters.html#compute-spectro-spatial-decomposition-ssd-spatial-filters

### PR DESCRIPTION
**Pull request by SHEscher**
#### Reference issue

Fixes #12807.

#### What does this implement/fix?

* Correct typo in SSD documentation
* minor language corrections 

#### Additional information

I have been asked by @larsoner to submit this PR.

### CODE REVIEW

**Review by drammock — APPROVED** [review]
LGTM. Could you add a changelog entry `doc/changes/devel/12942.bugfix.rst` please? You can find other `*bugfix.rst` examples in that folder, but the basic format is 

```
Fix typos in the Spatio-Spectral Decomposition example, by `Your Name`_.
```
You'll also need to add your name and URL to `doc/changes/names.inc` (alphabetically please)

**Comment by SHEscher** [comment]
Hey, I amended the former commit, adding the requested files and lines. 
Not sure whether amendments are your usual way to update PRs.

File: doc/changes/devel/12942.bugfix.rst
Comment (drammock) [inline doc/changes/devel/12942.bugfix.rst:1]: Number in the filename must match number of the PR (4 and 2 are transposed)

File: doc/changes/devel/12942.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix typos in the Spatio-Spectral Decomposition example, by :newcontrib:`Simon M. Hofmann`_.
```
Comment (drammock) [inline doc/changes/devel/12942.bugfix.rst:1]:
```suggestion
Fix typos in the Spatio-Spectral Decomposition example, by :newcontrib:`Simon M. Hofmann`.
```

When using the newcontrib role, the trailing underscore isn't needed

File: doc/changes/devel/12942.bugfix.rst
Comment (SHEscher) [reply doc/changes/devel/12942.bugfix.rst:1]: thanks! Should be done now

**Comment by larsoner** [comment]
Comments addressed and failures are unrelated (#12928), so I'll merge. Thanks @SHEscher !

**Comment by SHEscher** [comment]
> Comments addressed and failures are unrelated (#12928), so I'll merge. Thanks @SHEscher !

Thank you guys for your great work with the package!

**Pull request merged**